src/seat/mod.rs

```diff
@@ -91,11 +94,26 @@ impl SeatState {
         state
     }
 
+    /// Provide the [`CursorShapeManager`] to automatically handle server side cursor
+    /// when using [`Self::get_pointer_with_theme`] and [`Self::get_pointer_with_theme_and_data`].
+    pub fn with_cursor_shape_manager(
+        mut self,
+        cursor_shape_manager: Option<CursorShapeManager>,
+    ) -> Self {
+        self.cursor_shape_manager = cursor_shape_manager;
+        self
+    }
```

Comment:
Maybe we should lazily bind the manager internally when trying to get a pointer, so users will natively have a cursor shape working?

Reply:
Ya, I think that would work well in get_pointer_with theme.

Reply:
Should be fixed now.